### PR TITLE
Make it build with ghc 9.10

### PR DESCRIPTION
**Pull request by erikd**
# Changelog

```yaml
- description: |
    Make it build with ghc 9.10
# uncomment types applicable to the change:
  type:
  - maintenance    # not directly related to the code
```

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

**Review by carbolymer — APPROVED** [review]

File: cabal.project [hunk 1]
```diff
@@ -54,6 +54,21 @@ test-show-details: direct
...
+  -- These are required due to the above.
+  , cardano-ledger-alonzo:plutus-ledger-api
+  , cardano-ledger-alonzo-test:plutus-ledger-api
+  , cardano-ledger-babbage:plutus-ledger-api
+  , cardano-ledger-binary:plutus-ledger-api
+  , cardano-ledger-conway:plutus-ledger-api
+  , cardano-api:plutus-ledger-api
+  , cardano-api:prettyprinter-configurable
+  , cardano-api:base
+
```
Comment (erikd) [inline cabal.project]: I will let the node team decide whether the above needs to be fixed before this can be merged.

File: cabal.project [hunk 2]
```diff
@@ -54,6 +54,21 @@ test-show-details: direct
...
+allow-newer:
+  -- These are required due to the above.
+  , cardano-ledger-alonzo:plutus-ledger-api
+  , cardano-ledger-alonzo-test:plutus-ledger-api
+  , cardano-ledger-babbage:plutus-ledger-api
+  , cardano-ledger-binary:plutus-ledger-api
+  , cardano-ledger-conway:plutus-ledger-api
+  , cardano-api:plutus-ledger-api
+  , cardano-api:prettyprinter-configurable
+  , cardano-api:base
```
Comment (carbolymer) [inline cabal.project]: Those ones shouldn't be needed, right? Can't this be set in `cardano-api.cabal`?

Comment (erikd) [reply cabal.project]: To build with `ghc-9.10` they are at least until https://github.com/IntersectMBO/cardano-api/pull/599 is merged, but that is blocked by CI cross compile issues.

Comment (Jimbo4350) [reply cabal.project]: Are we in a rush for `9.10`? I'd prefer for the CI issues to be resolved in 599.

**Review by Jimbo4350 — CHANGES_REQUESTED** [review]
See comment

File: cabal.project (hunk 2 above)
Comment (erikd) [reply cabal.project]: Its not really urgent for anyone other than me. It is significantly more difficult for me to work on X that depends on Y if Y has not been merged to master.

File: .github/workflows/haskell.yml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        ghc: ["9.6", "9.8"]
+        ghc: ["9.6", "9.8", "9.10"]
```
Comment (carbolymer) [inline .github/workflows/haskell.yml:23]: If our next GHC target after 8.10 is 9.6, I'm wondering if we need both 9.8 and 9.10 here?

**Review by carbolymer — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**